MirrorProvider/MirrorProvider.Linux/LinuxFileSystemVirtualizer.cs

```diff
@@ -0,0 +1,247 @@
+        }
+
+        private Result OnGetFileStream(
+            ulong commandId,
+            string relativePath,
+            byte[] providerId,
+            byte[] contentId,
+            int triggeringProcessId,
+            string triggeringProcessName,
+            IntPtr fileHandle)
+        {
+            Console.WriteLine($"OnGetFileStream({commandId}, '{relativePath}', {contentId.Length}/{contentId[0]}:{contentId[1]}, {providerId.Length}/{providerId[0]}:{providerId[1]}, {triggeringProcessId}, {triggeringProcessName}, 0x{fileHandle.ToInt64():X})");
+
+            if (!this.FileExists(relativePath))
+            {
+                return Result.EFileNotFound;
+            }
+
+            try
+            {
+                const int bufferSize = 4096;
+                FileSystemResult hydrateFileResult = this.HydrateFile(
+                    relativePath,
+                    bufferSize,
+                    (buffer, bytesToCopy) =>
+                    {
+                        Result result = this.virtualizationInstance.WriteFileContents(
+                            fileHandle,
+                            buffer,
+                            (uint)bytesToCopy);
+                        if (result != Result.Success)
+                        {
+                        Console.WriteLine($"WriteFileContents failed: {result}");
+                            return false;
+                        }
+
+                        return true;
+                    });
+
+                if (hydrateFileResult != FileSystemResult.Success)
+                {
+                    return Result.EIOError;
+                }
+            }
+            catch (IOException e)
+            {
+                Console.WriteLine($"IOException in OnGetFileStream: {e.Message}");
+                return Result.EIOError;
+            }
+
+            return Result.Success;
+        }
+
+        private void OnFileModified(string relativePath)
+        {
+            Console.WriteLine($"OnFileModified: {relativePath}");
+        }
+
+        private Result OnPreDelete(string relativePath, bool isDirectory)
+        {
+            Console.WriteLine($"OnPreDelete (isDirectory: {isDirectory}): {relativePath}");
+            return Result.Success;
+        }
+
+        private void OnNewFileCreated(string relativePath, bool isDirectory)
+        {
+            Console.WriteLine($"OnNewFileCreated (isDirectory: {isDirectory}): {relativePath}");
+        }
+
+        private void OnFileRenamed(string relativeDestinationPath, bool isDirectory)
+        {
+            Console.WriteLine($"OnFileRenamed (isDirectory: {isDirectory}) destination: {relativeDestinationPath}");
+        }
+
+        private void OnHardLinkCreated(string relativeNewLinkPath)
+        {
+            Console.WriteLine($"OnHardLinkCreated: {relativeNewLinkPath}");
+        }
+
+        private bool TryGetSymLinkTarget(string relativePath, out string symLinkTarget)
```

Comment:
I noticed that this appears to be identical to `MacFileSystemVirtualizer.TryGetSymLinkTarget`.

Can `MacFileSystemVirtualizer.TryGetSymLinkTarget` be moved to a common spot rather than duplicating it here?